File: ExamPortalAPI/src/main/java/com/exam/services/QuestionService.java
package com.exam.services;

import java.util.Set;

import com.exam.dto.QuestionDto;
import com.exam.entities.Question;
import com.exam.exceptions.QuestionNotFoundException;
import com.exam.exceptions.QuizNotFoundException;

public interface QuestionService {
	public Question addQuestion(QuestionDto questionDto) throws QuizNotFoundException;

	public Question updateQuestion(Long id, QuestionDto questionDto)
			throws QuizNotFoundException, QuestionNotFoundException;

	public Set<Question> getQuestionsOfQuiz(Long quizId) throws QuizNotFoundException, QuestionNotFoundException;

	public Question getQuestion(Long questionId) throws QuestionNotFoundException;

	public void deleteQuestion(Long questionId) throws QuestionNotFoundException;
}

File: ExamPortalClient/src/app/services/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Question } from '../entities/question';
import { Helper } from '../helpers/helper';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {
  prefix: string = "question/"

  constructor(private http: HttpClient) { }

  getQuestionsOfQuizAdmin (quizId: number): Observable<any> {
    return this.http.get(Helper.baseUrl + `${this.prefix}get/quiz/all/${quizId}`);
  }

  createQuestion (question: Question): Observable<any> {
    return this.http.post(Helper.baseUrl + `${this.prefix}create`, question);
  }

  deleteQuestion (questionId: number): Observable<any> {
    return this.http.delete(Helper.baseUrl + `${this.prefix}delete/${questionId}`);
  }

  getQuestion (questionId: number): Observable<any> {
    return this.http.get(Helper.baseUrl + `${this.prefix}get/question/${questionId}`);
  }

  updateQuestion (questionId: number, question: Question): Observable<any> {
    return this.http.put(Helper.baseUrl + `${this.prefix}update/${questionId}`, question);
  }

}
